### ISSUE TITLE
Store and display three decimals for weight

### ISSUE THREAD

**Issue by drummer83** [openfoodfoundation/wishlist#442]
Product variants with a variant unit scale NOT being weight have an additional input to provide the weight of that variant (for packing/shipping reasons). The unit of this field is kg and it accepts decimals with three digits. However, after saving the data by clicking 'update' the value is rounded to two decimals and it is displayed with a trailing zero.

https://user-images.githubusercontent.com/1790176/234057583-bf2d57cc-ecf9-4a90-823e-37f9403135ab.mp4

## Which type of users does this problem affect (and how many, if known)?
- Hub Managers (MCFEs)
- Producers
- Instance Managers

## Potential solutions that will solve the problem
Store and display three decimals for the weight field.

## Connected wishlist and discovery discussions* [list precedent discussions]
#430 

## Additional context
Solving this was started in [#10684](https://github.com/openfoodfoundation/openfoodnetwork/pull/10684) but it didn't fix it completely.

**Comment by RachL** [comment]
Probably a papercut: not accept 3 digits for the weight attribute. Add an error message.

Accept 3 digits for weight: not that easy. Not a papercut.

workaround: if the user needs 3 digits they can also switch the unit to grams.

**Comment by RachL** [comment]
Yes but in another issue (i will fetch it) we plan to move this weight as a general field: even for product sold by weight or volume (as it's something needed for logistics).
